### ISSUE TITLE
Add small "real-world" protocol specification/example to the tutorial

### ISSUE THREAD

**Issue by sdleffler** [boltlabs-inc/dialectic#61]
Currently, the tutorial docs are very compartmentalized into sections for each operation/"instruction" type. It would be really nice to have a simple real-world protocol in the tutorial, at least as a specification using the `Session!` macro if not a fully working implementation. A couple candidates for potential protocols to use:

- SMTP (IETF [RFC 5321](https://tools.ietf.org/html/rfc5321))
- POP (IETF [RFC 1939](https://tools.ietf.org/html/rfc1939))

A full implementation would require a custom backend as well since these would have their own serialization mechanisms which would need to be built to their specifications.

**Comment by kwf** [comment]
Blocked on #98.
